pkg/process/monitor/process_monitor.go

```diff
@@ -238,23 +238,17 @@ func (pm *ProcessMonitor) Initialize() error {
 	var initErr error
 	pm.initOnce.Do(
 		func() {
-			pm.done = make(chan struct{})
-			pm.initCallbackRunner()
+			if err := pm.initNetlinkProcessEventMonitor(); err != nil {
+				initErr = err
+				return
+			}
 
-			pm.processMonitorWG.Add(1)
+			pm.initCallbackRunner()
 			// Setting up the main loop
-			pm.netlinkDoneChannel = make(chan struct{})
-			pm.netlinkErrorsChannel = make(chan error, 10)
-			pm.netlinkEventsChannel = make(chan netlink.ProcEvent, processMonitorEventQueueSize)
-
+			pm.done = make(chan struct{})
+			pm.processMonitorWG.Add(1)
```

Comment:
must be kept as first operation, otherwise we can have race conditions between initialize and stop